File: frontend/services/analytics.js
import api from './api';

const analyticsService = {
    trackPlay: async (postId) => {
        try {
            await api.post(`/analytics/track/play/${postId}`);
        } catch (e) {
            console.log("Analytics error (play)", e);
        }
    },

    trackProfileView: async (userId) => {
        try {
            await api.post(`/analytics/track/profile/${userId}`);
        } catch (e) {
            console.log("Analytics error (profile)", e);
        }
    }
};

export default analyticsService;

File: backend/app/api/v1/endpoints/analytics.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.api import deps
from app.models.models import Usuario, Publicacion

router = APIRouter()

@router.post("/track/play/{post_id}")
def track_play(post_id: int, db: Session = Depends(deps.get_db)):
    post = db.query(Publicacion).filter(Publicacion.id == post_id).first()
    if not post:
        raise HTTPException(status_code=404, detail="Post no encontrado")
    
    post.plays += 1
    db.commit()
    return {"status": "ok", "plays": post.plays}

@router.post("/track/profile/{user_id}")
def track_profile_view(user_id: int, db: Session = Depends(deps.get_db)):
    user = db.query(Usuario).filter(Usuario.id == user_id).first()
    if not user:
        raise HTTPException(status_code=404, detail="Usuario no encontrado")
    
    user.profile_views += 1
    db.commit()
    return {"status": "ok", "views": user.profile_views}
